### PR TITLE
feat: handle nested messages in fieldmask

### PR DESCRIPTION
**Pull request by ericwenn**
`Update` updates fields in destination message with values from source
message recursively.

### CODE REVIEW

File: fieldmask/update.go [hunk 1]
```diff
@@ -53,3 +53,74 @@ func UpdateTopLevelFields(mask *fieldmaskpb.FieldMask, dst, src proto.Message) {
...
+	}
+	if !isMessage(field) {
+		// not a message so can not have a field with that name
+		return
+	}
+	updateNamedField(dst.Get(field).Message(), src.Get(field).Message(), segments[1:])
+}
+
+func isMessage(field protoreflect.FieldDescriptor) bool {
+	return field.Kind() == protoreflect.MessageKind && !field.IsMap() && !field.IsList()
```
Comment (odsod) [inline fieldmask/update.go:125]: let's also handle GroupKind just for the hell of it?

Comment (odsod) [reply fieldmask/update.go:125]: not that *we* are using proto2 - but good to strive for covering the full scope of the proto spec IMO

Comment (ericwenn) [reply fieldmask/update.go:125]: Agreed. Still a bit confused about the different kinds tbh...

**Review by odsod — APPROVED** [review]
LGTM! Great tests

File: fieldmask/update.go (hunk 1 above)
Comment (odsod) [reply fieldmask/update.go:125]: Group is the one of the trippy ones, but it's basically a nested message baked into a single field.

https://developers.google.com/protocol-buffers/docs/reference/proto2-spec#group_field

Comment (odsod) [reply fieldmask/update.go:125]: So in almost all cases, groups can be treated as messages.

Comment (ericwenn) [reply fieldmask/update.go]: Thanks for link!

**Review by odsod — APPROVED** [review]

**Pull request merged**